var/spack/repos/builtin/packages/python/package.py

```diff
@@ -260,7 +260,7 @@ class Python(Package):
     variant("bz2", default=True, description="Build bz2 module")
     variant("lzma", default=True, description="Build lzma module")
     variant("pyexpat", default=True, description="Build pyexpat module")
-    variant("ctypes", default=True, description="Build ctypes module")
+    variant("ctypes", default=True, description="Build ctypes module", when="platform=win32")
```

Comment:
I'm still not sure about the Windows-specific changes. Maybe I'm reading the changelog incorrectly. Is ctypes always built (either with internal or external libffi) on macOS/Linux and optionally built (always external libffi) on Windows?

Reply:
With the current state of this PR, `ctypes` is always built on `linux/darwin` with external `libffi`. One can disable `ctypes` on windows, but the dependency on `libffi` still exists. 

On windows we set the `--no-ctypes` flag from `spack` based on the `ctypes` variant and I tried to preserve this functionality (but now `libffi` is always required and not optional). I tried to read the windows build scripts and I'm not well versed in MSBuild to understand what's going on there so I'm unsure if it is better to make the `libffi` dependency optional on windows. 

For reference: https://github.com/python/cpython/blob/main/PCbuild/build.bat

Reply:
@johnwparent is our Windows expert and probably wrote most of the Windows logic in the Python recipe, he may be able to comment.

Reply:
Right now we rely on Python's bundled libffi because `libffi` is an autotools package that doesn't currently build on Windows (and AFIACT wont). It cannot be optional as it stands.

There's an alternative in a different repo that is just a meson port of libffi that I've added as part of my campaign to make Gtk+ less of a nightmare on Windows, but that's not done yet. If you're willing to wait, I can PR what I have and we can try to get that in ASAP and then rebase this so you can use a Spack libffi.

Reply:
With https://github.com/spack/spack/pull/45803 merged, this PR can wait until your fix for windows builds of `libffi` is in.